robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,216 @@
+        :returns Base url for API request
+        """
+        _api_url = f'{self.rp_url}/api/v1/{self.rp_project}/'
+        return _api_url
+
+    @property
+    def headers(self):
+        """The headers for Report Requests.
+        :returns header for API request
+        """
+        _headers = {'Authorization': f'Bearer {self.rp_api_key}'}
+        return _headers
+
+
+class Launch(ReportPortal):
+    """Represents Launch in Report Portal
+
+    This holds the properties and function to interact with the
+    Report Portal launches or specific launch
+    """
+
+    def __init__(self):
+        """Initiate report portal launch properties"""
+        super(Launch, self).__init__()
+        self.launch_url = f'{self.api_url}/launch'
+
+    @property
+    def params(self):
+        """Parameters for Launch API request
+
+        :returns dict: parameters for API launch request
+        """
+        _params = {
+            'page.page': 1,
+            'page.size': 500,
+            'page.sort': 'start_time',
+        }
+        return _params
+
+    @property
+    def launches(self):
+        """Returns All launches in Report Portal
+
+        This does not includes each tests data for all tests in all launches.
+        But it includes just an overview data for all and each launch in Report Portal
+
+        :returns list: List of all launches in Report Portal
+        """
+        resp = requests.get(
+            self.launch_url, headers=self.headers, params=self.params, verify=False
+        )
+        resp.raise_for_status()
+        data = resp.json()
+        if data['content']:
+            launches = sorted(data['content'], key=lambda x: x['start_time'], reverse=True)
+            return launches
+
+    def launch(self, latest=1):
+        """Returns a specific launch overview data in Report Portal Project
+
+        This does not includes each tests data in launch
+
+        :param int latest: The launch place in Report portal
+            e.g To get the latest launch in Report Portal, set latest = 1,
+                For second latest, set latest = 2
+            By default, it returns the first/latest launch
+
+        :returns dict: The launch overview data of requested launch
+        """
+        latest = int(latest) if isinstance(latest, str) else latest
+        return self.launches[latest - 1]
+
+
+class Tests(Launch):
+    """Represents Tests in a Launch of a Report Portal"""
+
+    def __init__(self, launch):
+        """Initiates launch and properties for tests in that launch
+            e.g To get the latest launch in Report Portal, set latest = 1,
+                For second latest, set latest = 2
+
+        :param int launch: The launch position in Report portal
+        """
+        super(Tests, self).__init__()
+        self.tests_url = f'{self.api_url}/item'
+        self.launch = self.launch(latest=launch)
+
+    @property
+    def defect_types(self):
```

Comment:
Again this looks like a static class attribute or a module constant to me, not an instance property.

Reply:
Updated.